kie-server-parent/kie-server-tests/kie-server-integ-tests-optaplanner/src/test/java/org/kie/server/integrationtests/optaplanner/OptaplannerIntegrationTest.java

```diff
@@ -285,6 +297,42 @@ public void testExecuteSolver() throws Exception {
                                    SOLVER_1_ID);
     }
 
+    @Test
+    public void testPrometheusMetrics() throws Exception {
+        SolverInstance solverInstance = solverClient.createSolver(CONTAINER_1_ID,
+                SOLVER_PROMETHEUS_ID,
+                SOLVER_1_CONFIG);
+        assertNotNull(solverInstance);
+        assertEquals(SolverInstance.SolverStatus.NOT_SOLVING,
+                solverInstance.getStatus());
+
+        // the following status starts the solver
+        Object planningProblem = loadPlanningProblem(5,
+                15);
+        solverClient.solvePlanningProblem(CONTAINER_1_ID,
+                SOLVER_PROMETHEUS_ID,
+                planningProblem);
+
+        solverInstance = solverClient.getSolver(CONTAINER_1_ID,
+                SOLVER_PROMETHEUS_ID);
+
+        // solver should finish in 5 seconds, but we wait up to 15s before timing out
+        for (int i = 0; i < 5 && solverInstance.getStatus() == SolverInstance.SolverStatus.SOLVING; i++) {
+            Thread.sleep(3000);
```

Comment:
Thanks, the test should work, but I am a bit concerned about the time the test spends waiting on Thread.sleep(). Also it's not clear at the first glance what the timeout of the test is before someone reads your comment.

If I may offer an alternative solution:
```
final long SOLVER_STATUS_CHECK_PERIOD = 1000L;
while(solverInstance.getStatus() == SolverInstance.SolverStatus.SOLVING) {
 Thread.sleep(SOLVER_STATUS_CHECK_PERIOD);
}
```
and putting a timeout directly on the `@Test` annotation. What do you think about this approach?

Reply:
@rsynek I copied that hack from another test :-) and focused on this one. Should we correct both tests?

Reply:
@rsynek I updated my test only and checked for actual metric values published

Reply:
:) I know, I saw the test. If you want to improve the original test too, that would be great. If you find out that these tests differ just by (not)calling the `checkPrometheusMetrics()`, could be worth also reduce the duplication by introducing a new private method containing logic of the test.

If you find it too complicated, feel free to focus just on the tests related to Prometheus and let me know - I will file a Jira about the test refactoring.

Thanks!

Reply:
Yes, it is a bit different as it uses a different solver. But I can refactor that as well and reuse tests. I'll do that and we can create an additional JIRA for refactoring of these hardcoded 
timeouts

Reply:
Here it is @rsynek b75fb94e8081eb4e3c1273d2225761ac5be578ae I think my work on this PR is done now :-) LMK